app/api/files/signed-url/route.ts

```diff
@@ -0,0 +1,21 @@
+import { NextResponse } from "next/server";
+import supabase from "@/lib/supabase/serverClient";
+import { SUPABASE_STORAGE_BUCKET } from "@/lib/consts";
+
+export async function GET(req: Request) {
+  try {
+    const { searchParams } = new URL(req.url);
+    const key = searchParams.get("key");
+    if (!key) return NextResponse.json({ error: "Missing key" }, { status: 400 });
+
+    const { data, error } = await supabase.storage.from(SUPABASE_STORAGE_BUCKET).createSignedUrl(key, 300);
```

Comment:
Single responsibility principle
- actual: defining supabase libs in an API endpoint
- required: supabase libs stored in lib/supabase